pkg/api/v1/types.go

```diff
@@ -289,3 +292,99 @@ func NewVMReferenceFromName(name string) *VM {
 	vm.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
 	return vm
 }
+
+//TODO validate that this is correct
+func NewMigrationReferenceFromName(name string) *Migration {
+	migration := &Migration{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name,
+			Namespace: kubeapi.NamespaceDefault,
+			SelfLink:  fmt.Sprintf("/apis/%s/namespaces/%s/%s", GroupVersion.String(), kubeapi.NamespaceDefault, name),
+		},
+	}
+	migration.SetGroupVersionKind(schema.GroupVersionKind{Group: GroupVersion.Group, Kind: "VM", Version: GroupVersion.Version})
+	return migration
+}
+
+// A Migration is a job that moves a Virtual Machine from one node to another
+type Migration struct {
+	metav1.TypeMeta `json:",inline"`
+	ObjectMeta      v1.ObjectMeta   `json:"metadata,omitempty"`
+	Spec            MigrationSpec   `json:"spec,omitempty" valid:"required"`
+	Status          MigrationStatus `json:"message,omitempty"`
+}
+
+// MigrationSpec is a description of a VM Migration
+type MigrationSpec struct {
+	// The Kubernetes name of the Virtual Machine object to select for one migration.
+	// For example "destinationNodeName": "testvm" will migrate a VM called "testvm" in the namespace "default"
+	MigratingVMName string `json:"nodeName,omitempty"`
+	// Criteria to use when selecting the destination for the migration
+	// for example, to select by the hostname, specify `kubernetes.io/hostname: master`
+	// other possible choices include the hardware required to run the vm or
+	// or lableing of the nodes to indicate their roles in larger applications.
+	// examples:
+	// disktype: ssd,
+	// randomGenerator: /dev/random,
+	// randomGenerator: superfastdevice,
+	// app: mysql,
+	// licensedForServiceX: true
+	DestinationNodeSelector map[string]string `json:"nodeSelector,omitempty"`
+}
+
+type MigrationPhase string
+
+const (
+	// Migration has been scheduled but no update on the status has been recorded
+	MigrationPending MigrationPhase = "Pending"
+
+	// Migration is actively progressing
+	MigrationInProgress MigrationPhase = "In Progress"
```

Comment:
The VM is "Migrating" and the Migration is "In Progress."  The Migration is not Migrating....